// @openrouter/sdk/esm/models/operations/bulkunassignkeysfromguardrail.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5ff688d2a0f6
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BulkUnassignKeysFromGuardrailRequestBody$outboundSchema = z.object({
    keyHashes: z.array(z.string()),
}).transform((v) => {
    return remap$(v, {
        keyHashes: "key_hashes",
    });
});
export function bulkUnassignKeysFromGuardrailRequestBodyToJSON(bulkUnassignKeysFromGuardrailRequestBody) {
    return JSON.stringify(BulkUnassignKeysFromGuardrailRequestBody$outboundSchema.parse(bulkUnassignKeysFromGuardrailRequestBody));
}
/** @internal */
export const BulkUnassignKeysFromGuardrailRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    id: z.string(),
    requestBody: z.lazy(() => BulkUnassignKeysFromGuardrailRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
        requestBody: "RequestBody",
    });
});
export function bulkUnassignKeysFromGuardrailRequestToJSON(bulkUnassignKeysFromGuardrailRequest) {
    return JSON.stringify(BulkUnassignKeysFromGuardrailRequest$outboundSchema.parse(bulkUnassignKeysFromGuardrailRequest));
}
/** @internal */
export const BulkUnassignKeysFromGuardrailResponse$inboundSchema = z.object({
    unassigned_count: z.number(),
}).transform((v) => {
    return remap$(v, {
        "unassigned_count": "unassignedCount",
    });
});
export function bulkUnassignKeysFromGuardrailResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BulkUnassignKeysFromGuardrailResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BulkUnassignKeysFromGuardrailResponse' from JSON`);
}
