// @openrouter/sdk/esm/models/operations/bulkunassignmembersfromguardrail.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ab90e9780326
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BulkUnassignMembersFromGuardrailRequestBody$outboundSchema = z.object({
    memberUserIds: z.array(z.string()),
}).transform((v) => {
    return remap$(v, {
        memberUserIds: "member_user_ids",
    });
});
export function bulkUnassignMembersFromGuardrailRequestBodyToJSON(bulkUnassignMembersFromGuardrailRequestBody) {
    return JSON.stringify(BulkUnassignMembersFromGuardrailRequestBody$outboundSchema.parse(bulkUnassignMembersFromGuardrailRequestBody));
}
/** @internal */
export const BulkUnassignMembersFromGuardrailRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    id: z.string(),
    requestBody: z.lazy(() => BulkUnassignMembersFromGuardrailRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
        requestBody: "RequestBody",
    });
});
export function bulkUnassignMembersFromGuardrailRequestToJSON(bulkUnassignMembersFromGuardrailRequest) {
    return JSON.stringify(BulkUnassignMembersFromGuardrailRequest$outboundSchema.parse(bulkUnassignMembersFromGuardrailRequest));
}
/** @internal */
export const BulkUnassignMembersFromGuardrailResponse$inboundSchema = z.object({
    unassigned_count: z.number(),
}).transform((v) => {
    return remap$(v, {
        "unassigned_count": "unassignedCount",
    });
});
export function bulkUnassignMembersFromGuardrailResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BulkUnassignMembersFromGuardrailResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BulkUnassignMembersFromGuardrailResponse' from JSON`);
}
