// @openrouter/sdk/esm/models/operations/bulkassignkeystoguardrail.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 4f855e3ea589
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BulkAssignKeysToGuardrailRequestBody$outboundSchema = z.object({
    keyHashes: z.array(z.string()),
}).transform((v) => {
    return remap$(v, {
        keyHashes: "key_hashes",
    });
});
export function bulkAssignKeysToGuardrailRequestBodyToJSON(bulkAssignKeysToGuardrailRequestBody) {
    return JSON.stringify(BulkAssignKeysToGuardrailRequestBody$outboundSchema.parse(bulkAssignKeysToGuardrailRequestBody));
}
/** @internal */
export const BulkAssignKeysToGuardrailRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    id: z.string(),
    requestBody: z.lazy(() => BulkAssignKeysToGuardrailRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
        requestBody: "RequestBody",
    });
});
export function bulkAssignKeysToGuardrailRequestToJSON(bulkAssignKeysToGuardrailRequest) {
    return JSON.stringify(BulkAssignKeysToGuardrailRequest$outboundSchema.parse(bulkAssignKeysToGuardrailRequest));
}
/** @internal */
export const BulkAssignKeysToGuardrailResponse$inboundSchema = z.object({
    assigned_count: z.number(),
}).transform((v) => {
    return remap$(v, {
        "assigned_count": "assignedCount",
    });
});
export function bulkAssignKeysToGuardrailResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BulkAssignKeysToGuardrailResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BulkAssignKeysToGuardrailResponse' from JSON`);
}
