// @openrouter/sdk/esm/models/operations/bulkassignmemberstoguardrail.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 09a0f06fd33d
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const BulkAssignMembersToGuardrailRequestBody$outboundSchema = z.object({
    memberUserIds: z.array(z.string()),
}).transform((v) => {
    return remap$(v, {
        memberUserIds: "member_user_ids",
    });
});
export function bulkAssignMembersToGuardrailRequestBodyToJSON(bulkAssignMembersToGuardrailRequestBody) {
    return JSON.stringify(BulkAssignMembersToGuardrailRequestBody$outboundSchema.parse(bulkAssignMembersToGuardrailRequestBody));
}
/** @internal */
export const BulkAssignMembersToGuardrailRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    id: z.string(),
    requestBody: z.lazy(() => BulkAssignMembersToGuardrailRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
        requestBody: "RequestBody",
    });
});
export function bulkAssignMembersToGuardrailRequestToJSON(bulkAssignMembersToGuardrailRequest) {
    return JSON.stringify(BulkAssignMembersToGuardrailRequest$outboundSchema.parse(bulkAssignMembersToGuardrailRequest));
}
/** @internal */
export const BulkAssignMembersToGuardrailResponse$inboundSchema = z.object({
    assigned_count: z.number(),
}).transform((v) => {
    return remap$(v, {
        "assigned_count": "assignedCount",
    });
});
export function bulkAssignMembersToGuardrailResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BulkAssignMembersToGuardrailResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BulkAssignMembersToGuardrailResponse' from JSON`);
}
