// @openrouter/sdk/esm/models/operations/createguardrail.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5f0f56d14d79
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
/**
 * Interval at which the limit resets (daily, weekly, monthly)
 */
export const CreateGuardrailResetIntervalRequest = {
    Daily: "daily",
    Weekly: "weekly",
    Monthly: "monthly",
};
/**
 * Interval at which the limit resets (daily, weekly, monthly)
 */
export const CreateGuardrailResetIntervalResponse = {
    Daily: "daily",
    Weekly: "weekly",
    Monthly: "monthly",
};
/** @internal */
export const CreateGuardrailResetIntervalRequest$outboundSchema = openEnums.outboundSchema(CreateGuardrailResetIntervalRequest);
/** @internal */
export const CreateGuardrailRequestBody$outboundSchema = z.object({
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    limitUsd: z.nullable(z.number()).optional(),
    resetInterval: z.nullable(CreateGuardrailResetIntervalRequest$outboundSchema)
        .optional(),
    allowedProviders: z.nullable(z.array(z.string())).optional(),
    allowedModels: z.nullable(z.array(z.string())).optional(),
    enforceZdr: z.nullable(z.boolean()).optional(),
}).transform((v) => {
    return remap$(v, {
        limitUsd: "limit_usd",
        resetInterval: "reset_interval",
        allowedProviders: "allowed_providers",
        allowedModels: "allowed_models",
        enforceZdr: "enforce_zdr",
    });
});
export function createGuardrailRequestBodyToJSON(createGuardrailRequestBody) {
    return JSON.stringify(CreateGuardrailRequestBody$outboundSchema.parse(createGuardrailRequestBody));
}
/** @internal */
export const CreateGuardrailRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    requestBody: z.lazy(() => CreateGuardrailRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
        requestBody: "RequestBody",
    });
});
export function createGuardrailRequestToJSON(createGuardrailRequest) {
    return JSON.stringify(CreateGuardrailRequest$outboundSchema.parse(createGuardrailRequest));
}
/** @internal */
export const CreateGuardrailResetIntervalResponse$inboundSchema = openEnums.inboundSchema(CreateGuardrailResetIntervalResponse);
/** @internal */
export const CreateGuardrailData$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    limit_usd: z.nullable(z.number()).optional(),
    reset_interval: z.nullable(CreateGuardrailResetIntervalResponse$inboundSchema)
        .optional(),
    allowed_providers: z.nullable(z.array(z.string())).optional(),
    allowed_models: z.nullable(z.array(z.string())).optional(),
    enforce_zdr: z.nullable(z.boolean()).optional(),
    created_at: z.string(),
    updated_at: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "limit_usd": "limitUsd",
        "reset_interval": "resetInterval",
        "allowed_providers": "allowedProviders",
        "allowed_models": "allowedModels",
        "enforce_zdr": "enforceZdr",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
    });
});
export function createGuardrailDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateGuardrailData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateGuardrailData' from JSON`);
}
/** @internal */
export const CreateGuardrailResponse$inboundSchema = z.object({
    data: z.lazy(() => CreateGuardrailData$inboundSchema),
});
export function createGuardrailResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateGuardrailResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateGuardrailResponse' from JSON`);
}
