// @openrouter/sdk/esm/models/operations/deleteguardrail.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0319276942bf
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const DeleteGuardrailRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    id: z.string(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
    });
});
export function deleteGuardrailRequestToJSON(deleteGuardrailRequest) {
    return JSON.stringify(DeleteGuardrailRequest$outboundSchema.parse(deleteGuardrailRequest));
}
/** @internal */
export const DeleteGuardrailResponse$inboundSchema = z.object({
    deleted: z.literal(true),
});
export function deleteGuardrailResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteGuardrailResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteGuardrailResponse' from JSON`);
}
