// @openrouter/sdk/esm/models/operations/getgeneration.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5cdb2959d2a5
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
/**
 * Type of API used for the generation
 */
export const ApiType = {
    Completions: "completions",
    Embeddings: "embeddings",
};
export const ProviderName = {
    AnyScale: "AnyScale",
    Atoma: "Atoma",
    CentML: "Cent-ML",
    CrofAI: "CrofAI",
    Enfer: "Enfer",
    GoPomelo: "GoPomelo",
    HuggingFace: "HuggingFace",
    Hyperbolic2: "Hyperbolic 2",
    InoCloud: "InoCloud",
    Kluster: "Kluster",
    Lambda: "Lambda",
    Lepton: "Lepton",
    Lynn2: "Lynn 2",
    Lynn: "Lynn",
    Mancer: "Mancer",
    Meta: "Meta",
    Modal: "Modal",
    Nineteen: "Nineteen",
    OctoAI: "OctoAI",
    Recursal: "Recursal",
    Reflection: "Reflection",
    Replicate: "Replicate",
    SambaNova2: "SambaNova 2",
    SFCompute: "SF Compute",
    Targon: "Targon",
    Together2: "Together 2",
    Ubicloud: "Ubicloud",
    OneDotAI: "01.AI",
    Ai21: "AI21",
    AionLabs: "AionLabs",
    Alibaba: "Alibaba",
    Ambient: "Ambient",
    AmazonBedrock: "Amazon Bedrock",
    AmazonNova: "Amazon Nova",
    Anthropic: "Anthropic",
    ArceeAI: "Arcee AI",
    AtlasCloud: "AtlasCloud",
    Avian: "Avian",
    Azure: "Azure",
    BaseTen: "BaseTen",
    BytePlus: "BytePlus",
    BlackForestLabs: "Black Forest Labs",
    Cerebras: "Cerebras",
    Chutes: "Chutes",
    Cirrascale: "Cirrascale",
    Clarifai: "Clarifai",
    Cloudflare: "Cloudflare",
    Cohere: "Cohere",
    Crusoe: "Crusoe",
    DeepInfra: "DeepInfra",
    DeepSeek: "DeepSeek",
    Featherless: "Featherless",
    Fireworks: "Fireworks",
    Friendli: "Friendli",
    GMICloud: "GMICloud",
    Google: "Google",
    GoogleAIStudio: "Google AI Studio",
    Groq: "Groq",
    Hyperbolic: "Hyperbolic",
    Inception: "Inception",
    Inceptron: "Inceptron",
    InferenceNet: "InferenceNet",
    Infermatic: "Infermatic",
    Inflection: "Inflection",
    Liquid: "Liquid",
    Mara: "Mara",
    Mancer2: "Mancer 2",
    Minimax: "Minimax",
    ModelRun: "ModelRun",
    Mistral: "Mistral",
    Modular: "Modular",
    MoonshotAI: "Moonshot AI",
    Morph: "Morph",
    NCompass: "NCompass",
    Nebius: "Nebius",
    NextBit: "NextBit",
    Novita: "Novita",
    Nvidia: "Nvidia",
    OpenAI: "OpenAI",
    OpenInference: "OpenInference",
    Parasail: "Parasail",
    Perplexity: "Perplexity",
    Phala: "Phala",
    Relace: "Relace",
    SambaNova: "SambaNova",
    Seed: "Seed",
    SiliconFlow: "SiliconFlow",
    Sourceful: "Sourceful",
    StepFun: "StepFun",
    Stealth: "Stealth",
    StreamLake: "StreamLake",
    Switchpoint: "Switchpoint",
    Together: "Together",
    Upstage: "Upstage",
    Venice: "Venice",
    WandB: "WandB",
    Xiaomi: "Xiaomi",
    XAI: "xAI",
    ZAi: "Z.AI",
    FakeProvider: "FakeProvider",
};
/** @internal */
export const GetGenerationRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    id: z.string(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
    });
});
export function getGenerationRequestToJSON(getGenerationRequest) {
    return JSON.stringify(GetGenerationRequest$outboundSchema.parse(getGenerationRequest));
}
/** @internal */
export const ApiType$inboundSchema = openEnums
    .inboundSchema(ApiType);
/** @internal */
export const ProviderName$inboundSchema = openEnums.inboundSchema(ProviderName);
/** @internal */
export const ProviderResponse$inboundSchema = z.object({
    id: z.string().optional(),
    endpoint_id: z.string().optional(),
    model_permaslug: z.string().optional(),
    provider_name: ProviderName$inboundSchema.optional(),
    status: z.nullable(z.number()),
    latency: z.number().optional(),
    is_byok: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        "endpoint_id": "endpointId",
        "model_permaslug": "modelPermaslug",
        "provider_name": "providerName",
        "is_byok": "isByok",
    });
});
export function providerResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProviderResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProviderResponse' from JSON`);
}
/** @internal */
export const GetGenerationData$inboundSchema = z.object({
    id: z.string(),
    upstream_id: z.nullable(z.string()),
    total_cost: z.number(),
    cache_discount: z.nullable(z.number()),
    upstream_inference_cost: z.nullable(z.number()),
    created_at: z.string(),
    model: z.string(),
    app_id: z.nullable(z.number()),
    streamed: z.nullable(z.boolean()),
    cancelled: z.nullable(z.boolean()),
    provider_name: z.nullable(z.string()),
    latency: z.nullable(z.number()),
    moderation_latency: z.nullable(z.number()),
    generation_time: z.nullable(z.number()),
    finish_reason: z.nullable(z.string()),
    tokens_prompt: z.nullable(z.number()),
    tokens_completion: z.nullable(z.number()),
    native_tokens_prompt: z.nullable(z.number()),
    native_tokens_completion: z.nullable(z.number()),
    native_tokens_completion_images: z.nullable(z.number()),
    native_tokens_reasoning: z.nullable(z.number()),
    native_tokens_cached: z.nullable(z.number()),
    num_media_prompt: z.nullable(z.number()),
    num_input_audio_prompt: z.nullable(z.number()),
    num_media_completion: z.nullable(z.number()),
    num_search_results: z.nullable(z.number()),
    origin: z.string(),
    usage: z.number(),
    is_byok: z.boolean(),
    native_finish_reason: z.nullable(z.string()),
    external_user: z.nullable(z.string()),
    api_type: z.nullable(ApiType$inboundSchema),
    router: z.nullable(z.string()),
    provider_responses: z.nullable(z.array(z.lazy(() => ProviderResponse$inboundSchema))),
}).transform((v) => {
    return remap$(v, {
        "upstream_id": "upstreamId",
        "total_cost": "totalCost",
        "cache_discount": "cacheDiscount",
        "upstream_inference_cost": "upstreamInferenceCost",
        "created_at": "createdAt",
        "app_id": "appId",
        "provider_name": "providerName",
        "moderation_latency": "moderationLatency",
        "generation_time": "generationTime",
        "finish_reason": "finishReason",
        "tokens_prompt": "tokensPrompt",
        "tokens_completion": "tokensCompletion",
        "native_tokens_prompt": "nativeTokensPrompt",
        "native_tokens_completion": "nativeTokensCompletion",
        "native_tokens_completion_images": "nativeTokensCompletionImages",
        "native_tokens_reasoning": "nativeTokensReasoning",
        "native_tokens_cached": "nativeTokensCached",
        "num_media_prompt": "numMediaPrompt",
        "num_input_audio_prompt": "numInputAudioPrompt",
        "num_media_completion": "numMediaCompletion",
        "num_search_results": "numSearchResults",
        "is_byok": "isByok",
        "native_finish_reason": "nativeFinishReason",
        "external_user": "externalUser",
        "api_type": "apiType",
        "provider_responses": "providerResponses",
    });
});
export function getGenerationDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetGenerationData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetGenerationData' from JSON`);
}
/** @internal */
export const GetGenerationResponse$inboundSchema = z.object({
    data: z.lazy(() => GetGenerationData$inboundSchema),
});
export function getGenerationResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetGenerationResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetGenerationResponse' from JSON`);
}
