// @openrouter/sdk/esm/models/operations/getguardrail.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5a16015b4d1f
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
/**
 * Interval at which the limit resets (daily, weekly, monthly)
 */
export const GetGuardrailResetInterval = {
    Daily: "daily",
    Weekly: "weekly",
    Monthly: "monthly",
};
/** @internal */
export const GetGuardrailRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    id: z.string(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
    });
});
export function getGuardrailRequestToJSON(getGuardrailRequest) {
    return JSON.stringify(GetGuardrailRequest$outboundSchema.parse(getGuardrailRequest));
}
/** @internal */
export const GetGuardrailResetInterval$inboundSchema = openEnums.inboundSchema(GetGuardrailResetInterval);
/** @internal */
export const GetGuardrailData$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    limit_usd: z.nullable(z.number()).optional(),
    reset_interval: z.nullable(GetGuardrailResetInterval$inboundSchema)
        .optional(),
    allowed_providers: z.nullable(z.array(z.string())).optional(),
    allowed_models: z.nullable(z.array(z.string())).optional(),
    enforce_zdr: z.nullable(z.boolean()).optional(),
    created_at: z.string(),
    updated_at: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "limit_usd": "limitUsd",
        "reset_interval": "resetInterval",
        "allowed_providers": "allowedProviders",
        "allowed_models": "allowedModels",
        "enforce_zdr": "enforceZdr",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
    });
});
export function getGuardrailDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetGuardrailData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetGuardrailData' from JSON`);
}
/** @internal */
export const GetGuardrailResponse$inboundSchema = z.object({
    data: z.lazy(() => GetGuardrailData$inboundSchema),
});
export function getGuardrailResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetGuardrailResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetGuardrailResponse' from JSON`);
}
