// @openrouter/sdk/esm/models/operations/getuseractivity.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 19535b4511a4
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as models from "../index.js";
/** @internal */
export const GetUserActivityRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    date: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
    });
});
export function getUserActivityRequestToJSON(getUserActivityRequest) {
    return JSON.stringify(GetUserActivityRequest$outboundSchema.parse(getUserActivityRequest));
}
/** @internal */
export const GetUserActivityResponse$inboundSchema = z.object({
    data: z.array(models.ActivityItem$inboundSchema),
});
export function getUserActivityResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetUserActivityResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetUserActivityResponse' from JSON`);
}
