// @openrouter/sdk/esm/models/operations/listguardrailmemberassignments.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 37ee24c57c33
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ListGuardrailMemberAssignmentsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    id: z.string(),
    offset: z.string().optional(),
    limit: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
    });
});
export function listGuardrailMemberAssignmentsRequestToJSON(listGuardrailMemberAssignmentsRequest) {
    return JSON.stringify(ListGuardrailMemberAssignmentsRequest$outboundSchema.parse(listGuardrailMemberAssignmentsRequest));
}
/** @internal */
export const ListGuardrailMemberAssignmentsData$inboundSchema = z.object({
    id: z.string(),
    user_id: z.string(),
    organization_id: z.string(),
    guardrail_id: z.string(),
    assigned_by: z.nullable(z.string()),
    created_at: z.string(),
}).transform((v) => {
    return remap$(v, {
        "user_id": "userId",
        "organization_id": "organizationId",
        "guardrail_id": "guardrailId",
        "assigned_by": "assignedBy",
        "created_at": "createdAt",
    });
});
export function listGuardrailMemberAssignmentsDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListGuardrailMemberAssignmentsData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListGuardrailMemberAssignmentsData' from JSON`);
}
/** @internal */
export const ListGuardrailMemberAssignmentsResponse$inboundSchema = z.object({
    data: z.array(z.lazy(() => ListGuardrailMemberAssignmentsData$inboundSchema)),
    total_count: z.number(),
}).transform((v) => {
    return remap$(v, {
        "total_count": "totalCount",
    });
});
export function listGuardrailMemberAssignmentsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListGuardrailMemberAssignmentsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListGuardrailMemberAssignmentsResponse' from JSON`);
}
