// @openrouter/sdk/esm/models/operations/listguardrails.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 4c71cf4db173
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
/**
 * Interval at which the limit resets (daily, weekly, monthly)
 */
export const ListGuardrailsResetInterval = {
    Daily: "daily",
    Weekly: "weekly",
    Monthly: "monthly",
};
/** @internal */
export const ListGuardrailsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    offset: z.string().optional(),
    limit: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
    });
});
export function listGuardrailsRequestToJSON(listGuardrailsRequest) {
    return JSON.stringify(ListGuardrailsRequest$outboundSchema.parse(listGuardrailsRequest));
}
/** @internal */
export const ListGuardrailsResetInterval$inboundSchema = openEnums.inboundSchema(ListGuardrailsResetInterval);
/** @internal */
export const ListGuardrailsData$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    limit_usd: z.nullable(z.number()).optional(),
    reset_interval: z.nullable(ListGuardrailsResetInterval$inboundSchema)
        .optional(),
    allowed_providers: z.nullable(z.array(z.string())).optional(),
    allowed_models: z.nullable(z.array(z.string())).optional(),
    enforce_zdr: z.nullable(z.boolean()).optional(),
    created_at: z.string(),
    updated_at: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "limit_usd": "limitUsd",
        "reset_interval": "resetInterval",
        "allowed_providers": "allowedProviders",
        "allowed_models": "allowedModels",
        "enforce_zdr": "enforceZdr",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
    });
});
export function listGuardrailsDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListGuardrailsData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListGuardrailsData' from JSON`);
}
/** @internal */
export const ListGuardrailsResponse$inboundSchema = z.object({
    data: z.array(z.lazy(() => ListGuardrailsData$inboundSchema)),
    total_count: z.number(),
}).transform((v) => {
    return remap$(v, {
        "total_count": "totalCount",
    });
});
export function listGuardrailsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListGuardrailsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListGuardrailsResponse' from JSON`);
}
