// @openrouter/sdk/esm/models/operations/listmemberassignments.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7d02937e1a11
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ListMemberAssignmentsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    offset: z.string().optional(),
    limit: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
    });
});
export function listMemberAssignmentsRequestToJSON(listMemberAssignmentsRequest) {
    return JSON.stringify(ListMemberAssignmentsRequest$outboundSchema.parse(listMemberAssignmentsRequest));
}
/** @internal */
export const ListMemberAssignmentsData$inboundSchema = z.object({
    id: z.string(),
    user_id: z.string(),
    organization_id: z.string(),
    guardrail_id: z.string(),
    assigned_by: z.nullable(z.string()),
    created_at: z.string(),
}).transform((v) => {
    return remap$(v, {
        "user_id": "userId",
        "organization_id": "organizationId",
        "guardrail_id": "guardrailId",
        "assigned_by": "assignedBy",
        "created_at": "createdAt",
    });
});
export function listMemberAssignmentsDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListMemberAssignmentsData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListMemberAssignmentsData' from JSON`);
}
/** @internal */
export const ListMemberAssignmentsResponse$inboundSchema = z.object({
    data: z.array(z.lazy(() => ListMemberAssignmentsData$inboundSchema)),
    total_count: z.number(),
}).transform((v) => {
    return remap$(v, {
        "total_count": "totalCount",
    });
});
export function listMemberAssignmentsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListMemberAssignmentsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListMemberAssignmentsResponse' from JSON`);
}
