// @openrouter/sdk/esm/models/operations/listguardrailkeyassignments.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e616af13fd7a
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ListGuardrailKeyAssignmentsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    id: z.string(),
    offset: z.string().optional(),
    limit: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
    });
});
export function listGuardrailKeyAssignmentsRequestToJSON(listGuardrailKeyAssignmentsRequest) {
    return JSON.stringify(ListGuardrailKeyAssignmentsRequest$outboundSchema.parse(listGuardrailKeyAssignmentsRequest));
}
/** @internal */
export const ListGuardrailKeyAssignmentsData$inboundSchema = z.object({
    id: z.string(),
    key_hash: z.string(),
    guardrail_id: z.string(),
    key_name: z.string(),
    key_label: z.string(),
    assigned_by: z.nullable(z.string()),
    created_at: z.string(),
}).transform((v) => {
    return remap$(v, {
        "key_hash": "keyHash",
        "guardrail_id": "guardrailId",
        "key_name": "keyName",
        "key_label": "keyLabel",
        "assigned_by": "assignedBy",
        "created_at": "createdAt",
    });
});
export function listGuardrailKeyAssignmentsDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListGuardrailKeyAssignmentsData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListGuardrailKeyAssignmentsData' from JSON`);
}
/** @internal */
export const ListGuardrailKeyAssignmentsResponse$inboundSchema = z.object({
    data: z.array(z.lazy(() => ListGuardrailKeyAssignmentsData$inboundSchema)),
    total_count: z.number(),
}).transform((v) => {
    return remap$(v, {
        "total_count": "totalCount",
    });
});
export function listGuardrailKeyAssignmentsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListGuardrailKeyAssignmentsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListGuardrailKeyAssignmentsResponse' from JSON`);
}
