// @openrouter/sdk/esm/models/operations/listkeyassignments.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c4b0c0303428
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ListKeyAssignmentsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    offset: z.string().optional(),
    limit: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
    });
});
export function listKeyAssignmentsRequestToJSON(listKeyAssignmentsRequest) {
    return JSON.stringify(ListKeyAssignmentsRequest$outboundSchema.parse(listKeyAssignmentsRequest));
}
/** @internal */
export const ListKeyAssignmentsData$inboundSchema = z.object({
    id: z.string(),
    key_hash: z.string(),
    guardrail_id: z.string(),
    key_name: z.string(),
    key_label: z.string(),
    assigned_by: z.nullable(z.string()),
    created_at: z.string(),
}).transform((v) => {
    return remap$(v, {
        "key_hash": "keyHash",
        "guardrail_id": "guardrailId",
        "key_name": "keyName",
        "key_label": "keyLabel",
        "assigned_by": "assignedBy",
        "created_at": "createdAt",
    });
});
export function listKeyAssignmentsDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListKeyAssignmentsData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListKeyAssignmentsData' from JSON`);
}
/** @internal */
export const ListKeyAssignmentsResponse$inboundSchema = z.object({
    data: z.array(z.lazy(() => ListKeyAssignmentsData$inboundSchema)),
    total_count: z.number(),
}).transform((v) => {
    return remap$(v, {
        "total_count": "totalCount",
    });
});
export function listKeyAssignmentsResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListKeyAssignmentsResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListKeyAssignmentsResponse' from JSON`);
}
