// @openrouter/sdk/esm/models/operations/updateguardrail.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: d4106fbc6dca
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as openEnums from "../../types/enums.js";
/**
 * Interval at which the limit resets (daily, weekly, monthly)
 */
export const UpdateGuardrailResetIntervalRequest = {
    Daily: "daily",
    Weekly: "weekly",
    Monthly: "monthly",
};
/**
 * Interval at which the limit resets (daily, weekly, monthly)
 */
export const UpdateGuardrailResetIntervalResponse = {
    Daily: "daily",
    Weekly: "weekly",
    Monthly: "monthly",
};
/** @internal */
export const UpdateGuardrailResetIntervalRequest$outboundSchema = openEnums.outboundSchema(UpdateGuardrailResetIntervalRequest);
/** @internal */
export const UpdateGuardrailRequestBody$outboundSchema = z.object({
    name: z.string().optional(),
    description: z.nullable(z.string()).optional(),
    limitUsd: z.nullable(z.number()).optional(),
    resetInterval: z.nullable(UpdateGuardrailResetIntervalRequest$outboundSchema)
        .optional(),
    allowedProviders: z.nullable(z.array(z.string())).optional(),
    allowedModels: z.nullable(z.array(z.string())).optional(),
    enforceZdr: z.nullable(z.boolean()).optional(),
}).transform((v) => {
    return remap$(v, {
        limitUsd: "limit_usd",
        resetInterval: "reset_interval",
        allowedProviders: "allowed_providers",
        allowedModels: "allowed_models",
        enforceZdr: "enforce_zdr",
    });
});
export function updateGuardrailRequestBodyToJSON(updateGuardrailRequestBody) {
    return JSON.stringify(UpdateGuardrailRequestBody$outboundSchema.parse(updateGuardrailRequestBody));
}
/** @internal */
export const UpdateGuardrailRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    xTitle: z.string().optional(),
    id: z.string(),
    requestBody: z.lazy(() => UpdateGuardrailRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        xTitle: "X-Title",
        requestBody: "RequestBody",
    });
});
export function updateGuardrailRequestToJSON(updateGuardrailRequest) {
    return JSON.stringify(UpdateGuardrailRequest$outboundSchema.parse(updateGuardrailRequest));
}
/** @internal */
export const UpdateGuardrailResetIntervalResponse$inboundSchema = openEnums.inboundSchema(UpdateGuardrailResetIntervalResponse);
/** @internal */
export const UpdateGuardrailData$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    limit_usd: z.nullable(z.number()).optional(),
    reset_interval: z.nullable(UpdateGuardrailResetIntervalResponse$inboundSchema)
        .optional(),
    allowed_providers: z.nullable(z.array(z.string())).optional(),
    allowed_models: z.nullable(z.array(z.string())).optional(),
    enforce_zdr: z.nullable(z.boolean()).optional(),
    created_at: z.string(),
    updated_at: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "limit_usd": "limitUsd",
        "reset_interval": "resetInterval",
        "allowed_providers": "allowedProviders",
        "allowed_models": "allowedModels",
        "enforce_zdr": "enforceZdr",
        "created_at": "createdAt",
        "updated_at": "updatedAt",
    });
});
export function updateGuardrailDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateGuardrailData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateGuardrailData' from JSON`);
}
/** @internal */
export const UpdateGuardrailResponse$inboundSchema = z.object({
    data: z.lazy(() => UpdateGuardrailData$inboundSchema),
});
export function updateGuardrailResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateGuardrailResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateGuardrailResponse' from JSON`);
}
