// @openrouter/sdk/esm/models/assistantmessage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 2147a953e992
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ChatMessageContentItem$inboundSchema, ChatMessageContentItem$outboundSchema, } from "./chatmessagecontentitem.js";
import { ChatMessageToolCall$inboundSchema, ChatMessageToolCall$outboundSchema, } from "./chatmessagetoolcall.js";
import { Schema19$inboundSchema, Schema19$outboundSchema, } from "./schema19.js";
/** @internal */
export const AssistantMessageContent$inboundSchema = z.union([z.string(), z.array(ChatMessageContentItem$inboundSchema)]);
/** @internal */
export const AssistantMessageContent$outboundSchema = z.union([z.string(), z.array(ChatMessageContentItem$outboundSchema)]);
export function assistantMessageContentToJSON(assistantMessageContent) {
    return JSON.stringify(AssistantMessageContent$outboundSchema.parse(assistantMessageContent));
}
export function assistantMessageContentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AssistantMessageContent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AssistantMessageContent' from JSON`);
}
/** @internal */
export const AssistantMessageImageUrl$inboundSchema = z.object({
    url: z.string(),
});
/** @internal */
export const AssistantMessageImageUrl$outboundSchema = z.object({
    url: z.string(),
});
export function assistantMessageImageUrlToJSON(assistantMessageImageUrl) {
    return JSON.stringify(AssistantMessageImageUrl$outboundSchema.parse(assistantMessageImageUrl));
}
export function assistantMessageImageUrlFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AssistantMessageImageUrl$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AssistantMessageImageUrl' from JSON`);
}
/** @internal */
export const Image$inboundSchema = z.object({
    image_url: z.lazy(() => AssistantMessageImageUrl$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "image_url": "imageUrl",
    });
});
/** @internal */
export const Image$outboundSchema = z.object({
    imageUrl: z.lazy(() => AssistantMessageImageUrl$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        imageUrl: "image_url",
    });
});
export function imageToJSON(image) {
    return JSON.stringify(Image$outboundSchema.parse(image));
}
export function imageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Image$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Image' from JSON`);
}
/** @internal */
export const AssistantMessage$inboundSchema = z.object({
    role: z.literal("assistant"),
    content: z.nullable(z.union([z.string(), z.array(ChatMessageContentItem$inboundSchema)])).optional(),
    name: z.string().optional(),
    tool_calls: z.array(ChatMessageToolCall$inboundSchema).optional(),
    refusal: z.nullable(z.string()).optional(),
    reasoning: z.nullable(z.string()).optional(),
    reasoning_details: z.array(Schema19$inboundSchema).optional(),
    images: z.array(z.lazy(() => Image$inboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        "tool_calls": "toolCalls",
        "reasoning_details": "reasoningDetails",
    });
});
/** @internal */
export const AssistantMessage$outboundSchema = z.object({
    role: z.literal("assistant"),
    content: z.nullable(z.union([z.string(), z.array(ChatMessageContentItem$outboundSchema)])).optional(),
    name: z.string().optional(),
    toolCalls: z.array(ChatMessageToolCall$outboundSchema).optional(),
    refusal: z.nullable(z.string()).optional(),
    reasoning: z.nullable(z.string()).optional(),
    reasoningDetails: z.array(Schema19$outboundSchema).optional(),
    images: z.array(z.lazy(() => Image$outboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        toolCalls: "tool_calls",
        reasoningDetails: "reasoning_details",
    });
});
export function assistantMessageToJSON(assistantMessage) {
    return JSON.stringify(AssistantMessage$outboundSchema.parse(assistantMessage));
}
export function assistantMessageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AssistantMessage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AssistantMessage' from JSON`);
}
