// @openrouter/sdk/esm/models/chatgenerationparams.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f98a2a558f3f
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ChatStreamOptions$outboundSchema, } from "./chatstreamoptions.js";
import { Message$outboundSchema, } from "./message.js";
import { ProviderSortUnion$outboundSchema, } from "./providersortunion.js";
import { ReasoningSummaryVerbosity$outboundSchema, } from "./reasoningsummaryverbosity.js";
import { ResponseFormatJSONSchema$outboundSchema, } from "./responseformatjsonschema.js";
import { ResponseFormatTextGrammar$outboundSchema, } from "./responseformattextgrammar.js";
import { Schema10$outboundSchema, } from "./schema10.js";
import { Schema17$outboundSchema, } from "./schema17.js";
import { Schema5$outboundSchema, } from "./schema5.js";
import { ToolDefinitionJson$outboundSchema, } from "./tooldefinitionjson.js";
export const Schema3 = {
    Deny: "deny",
    Allow: "allow",
};
export const Schema8 = {
    Int4: "int4",
    Int8: "int8",
    Fp4: "fp4",
    Fp6: "fp6",
    Fp8: "fp8",
    Fp16: "fp16",
    Bf16: "bf16",
    Fp32: "fp32",
    Unknown: "unknown",
};
export const Route = {
    Fallback: "fallback",
    Sort: "sort",
};
export const Effort = {
    Xhigh: "xhigh",
    High: "high",
    Medium: "medium",
    Low: "low",
    Minimal: "minimal",
    None: "none",
};
export const Modality = {
    Text: "text",
    Image: "image",
};
/** @internal */
export const Schema3$outboundSchema = openEnums
    .outboundSchema(Schema3);
/** @internal */
export const Schema8$outboundSchema = openEnums
    .outboundSchema(Schema8);
/** @internal */
export const Schema15$outboundSchema = z
    .object({
    p50: z.nullable(z.number()).optional(),
    p75: z.nullable(z.number()).optional(),
    p90: z.nullable(z.number()).optional(),
    p99: z.nullable(z.number()).optional(),
});
export function schema15ToJSON(schema15) {
    return JSON.stringify(Schema15$outboundSchema.parse(schema15));
}
/** @internal */
export const Schema15Union$outboundSchema = z.union([z.number(), z.lazy(() => Schema15$outboundSchema)]);
export function schema15UnionToJSON(schema15Union) {
    return JSON.stringify(Schema15Union$outboundSchema.parse(schema15Union));
}
/** @internal */
export const Schema0$outboundSchema = z
    .object({
    allowFallbacks: z.nullable(z.boolean()).optional(),
    requireParameters: z.nullable(z.boolean()).optional(),
    dataCollection: z.nullable(Schema3$outboundSchema).optional(),
    zdr: z.nullable(z.boolean()).optional(),
    enforceDistillableText: z.nullable(z.boolean()).optional(),
    order: z.nullable(z.array(Schema5$outboundSchema)).optional(),
    only: z.nullable(z.array(Schema5$outboundSchema)).optional(),
    ignore: z.nullable(z.array(Schema5$outboundSchema)).optional(),
    quantizations: z.nullable(z.array(Schema8$outboundSchema)).optional(),
    sort: z.nullable(ProviderSortUnion$outboundSchema).optional(),
    maxPrice: Schema10$outboundSchema.optional(),
    preferredMinThroughput: z.nullable(z.union([z.number(), z.lazy(() => Schema15$outboundSchema)])).optional(),
    preferredMaxLatency: z.nullable(z.union([z.number(), z.lazy(() => Schema15$outboundSchema)])).optional(),
}).transform((v) => {
    return remap$(v, {
        allowFallbacks: "allow_fallbacks",
        requireParameters: "require_parameters",
        dataCollection: "data_collection",
        enforceDistillableText: "enforce_distillable_text",
        maxPrice: "max_price",
        preferredMinThroughput: "preferred_min_throughput",
        preferredMaxLatency: "preferred_max_latency",
    });
});
export function schema0ToJSON(schema0) {
    return JSON.stringify(Schema0$outboundSchema.parse(schema0));
}
/** @internal */
export const Route$outboundSchema = openEnums
    .outboundSchema(Route);
/** @internal */
export const Effort$outboundSchema = openEnums
    .outboundSchema(Effort);
/** @internal */
export const Reasoning$outboundSchema = z.object({
    effort: z.nullable(Effort$outboundSchema).optional(),
    summary: z.nullable(ReasoningSummaryVerbosity$outboundSchema).optional(),
});
export function reasoningToJSON(reasoning) {
    return JSON.stringify(Reasoning$outboundSchema.parse(reasoning));
}
/** @internal */
export const ResponseFormatPython$outboundSchema = z.object({
    type: z.literal("python"),
});
export function responseFormatPythonToJSON(responseFormatPython) {
    return JSON.stringify(ResponseFormatPython$outboundSchema.parse(responseFormatPython));
}
/** @internal */
export const ResponseFormatJSONObject$outboundSchema = z.object({
    type: z.literal("json_object"),
});
export function responseFormatJSONObjectToJSON(responseFormatJSONObject) {
    return JSON.stringify(ResponseFormatJSONObject$outboundSchema.parse(responseFormatJSONObject));
}
/** @internal */
export const ResponseFormatText$outboundSchema = z.object({
    type: z.literal("text"),
});
export function responseFormatTextToJSON(responseFormatText) {
    return JSON.stringify(ResponseFormatText$outboundSchema.parse(responseFormatText));
}
/** @internal */
export const ResponseFormat$outboundSchema = z.union([
    z.lazy(() => ResponseFormatText$outboundSchema),
    z.lazy(() => ResponseFormatJSONObject$outboundSchema),
    ResponseFormatJSONSchema$outboundSchema,
    ResponseFormatTextGrammar$outboundSchema,
    z.lazy(() => ResponseFormatPython$outboundSchema),
]);
export function responseFormatToJSON(responseFormat) {
    return JSON.stringify(ResponseFormat$outboundSchema.parse(responseFormat));
}
/** @internal */
export const Stop$outboundSchema = z.union([
    z.string(),
    z.array(z.string()),
]);
export function stopToJSON(stop) {
    return JSON.stringify(Stop$outboundSchema.parse(stop));
}
/** @internal */
export const Debug$outboundSchema = z.object({
    echoUpstreamBody: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        echoUpstreamBody: "echo_upstream_body",
    });
});
export function debugToJSON(debug) {
    return JSON.stringify(Debug$outboundSchema.parse(debug));
}
/** @internal */
export const ChatGenerationParamsImageConfig$outboundSchema = z.union([z.string(), z.number(), z.array(z.any())]);
export function chatGenerationParamsImageConfigToJSON(chatGenerationParamsImageConfig) {
    return JSON.stringify(ChatGenerationParamsImageConfig$outboundSchema.parse(chatGenerationParamsImageConfig));
}
/** @internal */
export const Modality$outboundSchema = openEnums
    .outboundSchema(Modality);
/** @internal */
export const ChatGenerationParams$outboundSchema = z.object({
    provider: z.nullable(z.lazy(() => Schema0$outboundSchema)).optional(),
    plugins: z.array(Schema17$outboundSchema).optional(),
    route: z.nullable(Route$outboundSchema).optional(),
    user: z.string().optional(),
    sessionId: z.string().optional(),
    messages: z.array(Message$outboundSchema),
    model: z.string().optional(),
    models: z.array(z.string()).optional(),
    frequencyPenalty: z.nullable(z.number()).optional(),
    logitBias: z.nullable(z.record(z.string(), z.number())).optional(),
    logprobs: z.nullable(z.boolean()).optional(),
    topLogprobs: z.nullable(z.number()).optional(),
    maxCompletionTokens: z.nullable(z.number()).optional(),
    maxTokens: z.nullable(z.number()).optional(),
    metadata: z.record(z.string(), z.string()).optional(),
    presencePenalty: z.nullable(z.number()).optional(),
    reasoning: z.lazy(() => Reasoning$outboundSchema).optional(),
    responseFormat: z.union([
        z.lazy(() => ResponseFormatText$outboundSchema),
        z.lazy(() => ResponseFormatJSONObject$outboundSchema),
        ResponseFormatJSONSchema$outboundSchema,
        ResponseFormatTextGrammar$outboundSchema,
        z.lazy(() => ResponseFormatPython$outboundSchema),
    ]).optional(),
    seed: z.nullable(z.int()).optional(),
    stop: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    stream: z.boolean().default(false),
    streamOptions: z.nullable(ChatStreamOptions$outboundSchema).optional(),
    temperature: z.nullable(z.number()).optional(),
    toolChoice: z.any().optional(),
    tools: z.array(ToolDefinitionJson$outboundSchema).optional(),
    topP: z.nullable(z.number()).optional(),
    debug: z.lazy(() => Debug$outboundSchema).optional(),
    imageConfig: z.record(z.string(), z.union([z.string(), z.number(), z.array(z.any())])).optional(),
    modalities: z.array(Modality$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        sessionId: "session_id",
        frequencyPenalty: "frequency_penalty",
        logitBias: "logit_bias",
        topLogprobs: "top_logprobs",
        maxCompletionTokens: "max_completion_tokens",
        maxTokens: "max_tokens",
        presencePenalty: "presence_penalty",
        responseFormat: "response_format",
        streamOptions: "stream_options",
        toolChoice: "tool_choice",
        topP: "top_p",
        imageConfig: "image_config",
    });
});
export function chatGenerationParamsToJSON(chatGenerationParams) {
    return JSON.stringify(ChatGenerationParams$outboundSchema.parse(chatGenerationParams));
}
