// @openrouter/sdk/esm/models/openresponsesinputmessageitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: cccc4eb21a9b
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ResponseInputAudio$outboundSchema, } from "./responseinputaudio.js";
import { ResponseInputFile$outboundSchema, } from "./responseinputfile.js";
import { ResponseInputText$outboundSchema, } from "./responseinputtext.js";
import { ResponseInputVideo$outboundSchema, } from "./responseinputvideo.js";
export const OpenResponsesInputMessageItemTypeMessage = {
    Message: "message",
};
export const OpenResponsesInputMessageItemRoleDeveloper = {
    Developer: "developer",
};
export const OpenResponsesInputMessageItemRoleSystem = {
    System: "system",
};
export const OpenResponsesInputMessageItemRoleUser = {
    User: "user",
};
export const OpenResponsesInputMessageItemDetail = {
    Auto: "auto",
    High: "high",
    Low: "low",
};
/** @internal */
export const OpenResponsesInputMessageItemTypeMessage$outboundSchema = z.enum(OpenResponsesInputMessageItemTypeMessage);
/** @internal */
export const OpenResponsesInputMessageItemRoleDeveloper$outboundSchema = z.enum(OpenResponsesInputMessageItemRoleDeveloper);
/** @internal */
export const OpenResponsesInputMessageItemRoleSystem$outboundSchema = z.enum(OpenResponsesInputMessageItemRoleSystem);
/** @internal */
export const OpenResponsesInputMessageItemRoleUser$outboundSchema = z.enum(OpenResponsesInputMessageItemRoleUser);
/** @internal */
export const OpenResponsesInputMessageItemRoleUnion$outboundSchema = z.union([
    OpenResponsesInputMessageItemRoleUser$outboundSchema,
    OpenResponsesInputMessageItemRoleSystem$outboundSchema,
    OpenResponsesInputMessageItemRoleDeveloper$outboundSchema,
]);
export function openResponsesInputMessageItemRoleUnionToJSON(openResponsesInputMessageItemRoleUnion) {
    return JSON.stringify(OpenResponsesInputMessageItemRoleUnion$outboundSchema.parse(openResponsesInputMessageItemRoleUnion));
}
/** @internal */
export const OpenResponsesInputMessageItemDetail$outboundSchema = openEnums.outboundSchema(OpenResponsesInputMessageItemDetail);
/** @internal */
export const OpenResponsesInputMessageItemContentInputImage$outboundSchema = z.object({
    type: z.literal("input_image"),
    detail: OpenResponsesInputMessageItemDetail$outboundSchema,
    imageUrl: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        imageUrl: "image_url",
    });
});
export function openResponsesInputMessageItemContentInputImageToJSON(openResponsesInputMessageItemContentInputImage) {
    return JSON.stringify(OpenResponsesInputMessageItemContentInputImage$outboundSchema.parse(openResponsesInputMessageItemContentInputImage));
}
/** @internal */
export const OpenResponsesInputMessageItemContentUnion$outboundSchema = z.union([
    ResponseInputText$outboundSchema,
    z.lazy(() => OpenResponsesInputMessageItemContentInputImage$outboundSchema),
    ResponseInputFile$outboundSchema,
    ResponseInputAudio$outboundSchema,
    ResponseInputVideo$outboundSchema,
]);
export function openResponsesInputMessageItemContentUnionToJSON(openResponsesInputMessageItemContentUnion) {
    return JSON.stringify(OpenResponsesInputMessageItemContentUnion$outboundSchema.parse(openResponsesInputMessageItemContentUnion));
}
/** @internal */
export const OpenResponsesInputMessageItem$outboundSchema = z.object({
    id: z.string().optional(),
    type: OpenResponsesInputMessageItemTypeMessage$outboundSchema.optional(),
    role: z.union([
        OpenResponsesInputMessageItemRoleUser$outboundSchema,
        OpenResponsesInputMessageItemRoleSystem$outboundSchema,
        OpenResponsesInputMessageItemRoleDeveloper$outboundSchema,
    ]),
    content: z.array(z.union([
        ResponseInputText$outboundSchema,
        z.lazy(() => OpenResponsesInputMessageItemContentInputImage$outboundSchema),
        ResponseInputFile$outboundSchema,
        ResponseInputAudio$outboundSchema,
        ResponseInputVideo$outboundSchema,
    ])),
});
export function openResponsesInputMessageItemToJSON(openResponsesInputMessageItem) {
    return JSON.stringify(OpenResponsesInputMessageItem$outboundSchema.parse(openResponsesInputMessageItem));
}
