// @openrouter/sdk/esm/models/openresponseseasyinputmessage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 9cae543ceb47
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ResponseInputAudio$outboundSchema, } from "./responseinputaudio.js";
import { ResponseInputFile$outboundSchema, } from "./responseinputfile.js";
import { ResponseInputText$outboundSchema, } from "./responseinputtext.js";
import { ResponseInputVideo$outboundSchema, } from "./responseinputvideo.js";
export const OpenResponsesEasyInputMessageTypeMessage = {
    Message: "message",
};
export const OpenResponsesEasyInputMessageRoleDeveloper = {
    Developer: "developer",
};
export const OpenResponsesEasyInputMessageRoleAssistant = {
    Assistant: "assistant",
};
export const OpenResponsesEasyInputMessageRoleSystem = {
    System: "system",
};
export const OpenResponsesEasyInputMessageRoleUser = {
    User: "user",
};
export const OpenResponsesEasyInputMessageDetail = {
    Auto: "auto",
    High: "high",
    Low: "low",
};
/** @internal */
export const OpenResponsesEasyInputMessageTypeMessage$outboundSchema = z.enum(OpenResponsesEasyInputMessageTypeMessage);
/** @internal */
export const OpenResponsesEasyInputMessageRoleDeveloper$outboundSchema = z.enum(OpenResponsesEasyInputMessageRoleDeveloper);
/** @internal */
export const OpenResponsesEasyInputMessageRoleAssistant$outboundSchema = z.enum(OpenResponsesEasyInputMessageRoleAssistant);
/** @internal */
export const OpenResponsesEasyInputMessageRoleSystem$outboundSchema = z.enum(OpenResponsesEasyInputMessageRoleSystem);
/** @internal */
export const OpenResponsesEasyInputMessageRoleUser$outboundSchema = z.enum(OpenResponsesEasyInputMessageRoleUser);
/** @internal */
export const OpenResponsesEasyInputMessageRoleUnion$outboundSchema = z.union([
    OpenResponsesEasyInputMessageRoleUser$outboundSchema,
    OpenResponsesEasyInputMessageRoleSystem$outboundSchema,
    OpenResponsesEasyInputMessageRoleAssistant$outboundSchema,
    OpenResponsesEasyInputMessageRoleDeveloper$outboundSchema,
]);
export function openResponsesEasyInputMessageRoleUnionToJSON(openResponsesEasyInputMessageRoleUnion) {
    return JSON.stringify(OpenResponsesEasyInputMessageRoleUnion$outboundSchema.parse(openResponsesEasyInputMessageRoleUnion));
}
/** @internal */
export const OpenResponsesEasyInputMessageDetail$outboundSchema = openEnums.outboundSchema(OpenResponsesEasyInputMessageDetail);
/** @internal */
export const OpenResponsesEasyInputMessageContentInputImage$outboundSchema = z.object({
    type: z.literal("input_image"),
    detail: OpenResponsesEasyInputMessageDetail$outboundSchema,
    imageUrl: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        imageUrl: "image_url",
    });
});
export function openResponsesEasyInputMessageContentInputImageToJSON(openResponsesEasyInputMessageContentInputImage) {
    return JSON.stringify(OpenResponsesEasyInputMessageContentInputImage$outboundSchema.parse(openResponsesEasyInputMessageContentInputImage));
}
/** @internal */
export const OpenResponsesEasyInputMessageContentUnion1$outboundSchema = z.union([
    ResponseInputText$outboundSchema,
    z.lazy(() => OpenResponsesEasyInputMessageContentInputImage$outboundSchema),
    ResponseInputFile$outboundSchema,
    ResponseInputAudio$outboundSchema,
    ResponseInputVideo$outboundSchema,
]);
export function openResponsesEasyInputMessageContentUnion1ToJSON(openResponsesEasyInputMessageContentUnion1) {
    return JSON.stringify(OpenResponsesEasyInputMessageContentUnion1$outboundSchema.parse(openResponsesEasyInputMessageContentUnion1));
}
/** @internal */
export const OpenResponsesEasyInputMessageContentUnion2$outboundSchema = z.union([
    z.array(z.union([
        ResponseInputText$outboundSchema,
        z.lazy(() => OpenResponsesEasyInputMessageContentInputImage$outboundSchema),
        ResponseInputFile$outboundSchema,
        ResponseInputAudio$outboundSchema,
        ResponseInputVideo$outboundSchema,
    ])),
    z.string(),
]);
export function openResponsesEasyInputMessageContentUnion2ToJSON(openResponsesEasyInputMessageContentUnion2) {
    return JSON.stringify(OpenResponsesEasyInputMessageContentUnion2$outboundSchema.parse(openResponsesEasyInputMessageContentUnion2));
}
/** @internal */
export const OpenResponsesEasyInputMessage$outboundSchema = z.object({
    type: OpenResponsesEasyInputMessageTypeMessage$outboundSchema.optional(),
    role: z.union([
        OpenResponsesEasyInputMessageRoleUser$outboundSchema,
        OpenResponsesEasyInputMessageRoleSystem$outboundSchema,
        OpenResponsesEasyInputMessageRoleAssistant$outboundSchema,
        OpenResponsesEasyInputMessageRoleDeveloper$outboundSchema,
    ]),
    content: z.union([
        z.array(z.union([
            ResponseInputText$outboundSchema,
            z.lazy(() => OpenResponsesEasyInputMessageContentInputImage$outboundSchema),
            ResponseInputFile$outboundSchema,
            ResponseInputAudio$outboundSchema,
            ResponseInputVideo$outboundSchema,
        ])),
        z.string(),
    ]),
});
export function openResponsesEasyInputMessageToJSON(openResponsesEasyInputMessage) {
    return JSON.stringify(OpenResponsesEasyInputMessage$outboundSchema.parse(openResponsesEasyInputMessage));
}
