// @openrouter/sdk/esm/models/responsesoutputitemreasoning.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 191f7f61ea84
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { ReasoningSummaryText$inboundSchema, ReasoningSummaryText$outboundSchema, } from "./reasoningsummarytext.js";
import { ReasoningTextContent$inboundSchema, ReasoningTextContent$outboundSchema, } from "./reasoningtextcontent.js";
export const ResponsesOutputItemReasoningType = {
    Reasoning: "reasoning",
};
export const ResponsesOutputItemReasoningStatusInProgress = {
    InProgress: "in_progress",
};
export const ResponsesOutputItemReasoningStatusIncomplete = {
    Incomplete: "incomplete",
};
export const ResponsesOutputItemReasoningStatusCompleted = {
    Completed: "completed",
};
/**
 * The format of the reasoning content
 */
export const ResponsesOutputItemReasoningFormat = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    AzureOpenaiResponsesV1: "azure-openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
/** @internal */
export const ResponsesOutputItemReasoningType$inboundSchema = z.enum(ResponsesOutputItemReasoningType);
/** @internal */
export const ResponsesOutputItemReasoningType$outboundSchema = ResponsesOutputItemReasoningType$inboundSchema;
/** @internal */
export const ResponsesOutputItemReasoningStatusInProgress$inboundSchema = z.enum(ResponsesOutputItemReasoningStatusInProgress);
/** @internal */
export const ResponsesOutputItemReasoningStatusInProgress$outboundSchema = ResponsesOutputItemReasoningStatusInProgress$inboundSchema;
/** @internal */
export const ResponsesOutputItemReasoningStatusIncomplete$inboundSchema = z.enum(ResponsesOutputItemReasoningStatusIncomplete);
/** @internal */
export const ResponsesOutputItemReasoningStatusIncomplete$outboundSchema = ResponsesOutputItemReasoningStatusIncomplete$inboundSchema;
/** @internal */
export const ResponsesOutputItemReasoningStatusCompleted$inboundSchema = z.enum(ResponsesOutputItemReasoningStatusCompleted);
/** @internal */
export const ResponsesOutputItemReasoningStatusCompleted$outboundSchema = ResponsesOutputItemReasoningStatusCompleted$inboundSchema;
/** @internal */
export const ResponsesOutputItemReasoningStatusUnion$inboundSchema = z.union([
    ResponsesOutputItemReasoningStatusCompleted$inboundSchema,
    ResponsesOutputItemReasoningStatusIncomplete$inboundSchema,
    ResponsesOutputItemReasoningStatusInProgress$inboundSchema,
]);
/** @internal */
export const ResponsesOutputItemReasoningStatusUnion$outboundSchema = z.union([
    ResponsesOutputItemReasoningStatusCompleted$outboundSchema,
    ResponsesOutputItemReasoningStatusIncomplete$outboundSchema,
    ResponsesOutputItemReasoningStatusInProgress$outboundSchema,
]);
export function responsesOutputItemReasoningStatusUnionToJSON(responsesOutputItemReasoningStatusUnion) {
    return JSON.stringify(ResponsesOutputItemReasoningStatusUnion$outboundSchema.parse(responsesOutputItemReasoningStatusUnion));
}
export function responsesOutputItemReasoningStatusUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputItemReasoningStatusUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputItemReasoningStatusUnion' from JSON`);
}
/** @internal */
export const ResponsesOutputItemReasoningFormat$inboundSchema = openEnums.inboundSchema(ResponsesOutputItemReasoningFormat);
/** @internal */
export const ResponsesOutputItemReasoningFormat$outboundSchema = openEnums.outboundSchema(ResponsesOutputItemReasoningFormat);
/** @internal */
export const ResponsesOutputItemReasoning$inboundSchema = z.object({
    type: ResponsesOutputItemReasoningType$inboundSchema,
    id: z.string(),
    content: z.array(ReasoningTextContent$inboundSchema).optional(),
    summary: z.array(ReasoningSummaryText$inboundSchema),
    encrypted_content: z.nullable(z.string()).optional(),
    status: z.union([
        ResponsesOutputItemReasoningStatusCompleted$inboundSchema,
        ResponsesOutputItemReasoningStatusIncomplete$inboundSchema,
        ResponsesOutputItemReasoningStatusInProgress$inboundSchema,
    ]).optional(),
    signature: z.nullable(z.string()).optional(),
    format: z.nullable(ResponsesOutputItemReasoningFormat$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "encrypted_content": "encryptedContent",
    });
});
/** @internal */
export const ResponsesOutputItemReasoning$outboundSchema = z.object({
    type: ResponsesOutputItemReasoningType$outboundSchema,
    id: z.string(),
    content: z.array(ReasoningTextContent$outboundSchema).optional(),
    summary: z.array(ReasoningSummaryText$outboundSchema),
    encryptedContent: z.nullable(z.string()).optional(),
    status: z.union([
        ResponsesOutputItemReasoningStatusCompleted$outboundSchema,
        ResponsesOutputItemReasoningStatusIncomplete$outboundSchema,
        ResponsesOutputItemReasoningStatusInProgress$outboundSchema,
    ]).optional(),
    signature: z.nullable(z.string()).optional(),
    format: z.nullable(ResponsesOutputItemReasoningFormat$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        encryptedContent: "encrypted_content",
    });
});
export function responsesOutputItemReasoningToJSON(responsesOutputItemReasoning) {
    return JSON.stringify(ResponsesOutputItemReasoning$outboundSchema.parse(responsesOutputItemReasoning));
}
export function responsesOutputItemReasoningFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponsesOutputItemReasoning$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponsesOutputItemReasoning' from JSON`);
}
