// @openrouter/sdk/esm/models/schema10.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3a361cbac0f8
 */
import * as z from "zod/v4";
import { Schema14$outboundSchema, } from "./schema14.js";
/** @internal */
export const Prompt$outboundSchema = z
    .union([z.number(), z.string(), z.any()]);
export function promptToJSON(prompt) {
    return JSON.stringify(Prompt$outboundSchema.parse(prompt));
}
/** @internal */
export const Completion$outboundSchema = z.union([z.number(), z.string(), z.any()]);
export function completionToJSON(completion) {
    return JSON.stringify(Completion$outboundSchema.parse(completion));
}
/** @internal */
export const Schema10$outboundSchema = z
    .object({
    prompt: z.union([z.number(), z.string(), z.any()]).optional(),
    completion: z.union([z.number(), z.string(), z.any()]).optional(),
    image: Schema14$outboundSchema.optional(),
    audio: Schema14$outboundSchema.optional(),
    request: Schema14$outboundSchema.optional(),
});
export function schema10ToJSON(schema10) {
    return JSON.stringify(Schema10$outboundSchema.parse(schema10));
}
