// @openrouter/sdk/esm/models/schema17.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f882e1dc50da
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
export const PdfEngine = {
    MistralOcr: "mistral-ocr",
    PdfText: "pdf-text",
    Native: "native",
};
export const Engine = {
    Native: "native",
    Exa: "exa",
};
/** @internal */
export const Schema17ResponseHealing$outboundSchema = z.object({
    id: z.literal("response-healing"),
    enabled: z.boolean().optional(),
});
export function schema17ResponseHealingToJSON(schema17ResponseHealing) {
    return JSON.stringify(Schema17ResponseHealing$outboundSchema.parse(schema17ResponseHealing));
}
/** @internal */
export const PdfEngine$outboundSchema = openEnums
    .outboundSchema(PdfEngine);
/** @internal */
export const Pdf$outboundSchema = z.object({
    engine: PdfEngine$outboundSchema.optional(),
});
export function pdfToJSON(pdf) {
    return JSON.stringify(Pdf$outboundSchema.parse(pdf));
}
/** @internal */
export const Schema17FileParser$outboundSchema = z.object({
    id: z.literal("file-parser"),
    enabled: z.boolean().optional(),
    pdf: z.lazy(() => Pdf$outboundSchema).optional(),
});
export function schema17FileParserToJSON(schema17FileParser) {
    return JSON.stringify(Schema17FileParser$outboundSchema.parse(schema17FileParser));
}
/** @internal */
export const Engine$outboundSchema = openEnums
    .outboundSchema(Engine);
/** @internal */
export const Schema17Web$outboundSchema = z.object({
    id: z.literal("web"),
    enabled: z.boolean().optional(),
    maxResults: z.number().optional(),
    searchPrompt: z.string().optional(),
    engine: Engine$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        searchPrompt: "search_prompt",
    });
});
export function schema17WebToJSON(schema17Web) {
    return JSON.stringify(Schema17Web$outboundSchema.parse(schema17Web));
}
/** @internal */
export const Schema17Moderation$outboundSchema = z.object({
    id: z.literal("moderation"),
});
export function schema17ModerationToJSON(schema17Moderation) {
    return JSON.stringify(Schema17Moderation$outboundSchema.parse(schema17Moderation));
}
/** @internal */
export const Schema17AutoRouter$outboundSchema = z.object({
    id: z.literal("auto-router"),
    enabled: z.boolean().optional(),
    allowedModels: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        allowedModels: "allowed_models",
    });
});
export function schema17AutoRouterToJSON(schema17AutoRouter) {
    return JSON.stringify(Schema17AutoRouter$outboundSchema.parse(schema17AutoRouter));
}
/** @internal */
export const Schema17$outboundSchema = z
    .union([
    z.lazy(() => Schema17AutoRouter$outboundSchema),
    z.lazy(() => Schema17Moderation$outboundSchema),
    z.lazy(() => Schema17Web$outboundSchema),
    z.lazy(() => Schema17FileParser$outboundSchema),
    z.lazy(() => Schema17ResponseHealing$outboundSchema),
]);
export function schema17ToJSON(schema17) {
    return JSON.stringify(Schema17$outboundSchema.parse(schema17));
}
