// @openrouter/sdk/esm/models/schema19.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 81bc92589adb
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const Schema21 = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    AzureOpenaiResponsesV1: "azure-openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
/** @internal */
export const Schema21$inboundSchema = openEnums
    .inboundSchema(Schema21);
/** @internal */
export const Schema21$outboundSchema = openEnums
    .outboundSchema(Schema21);
/** @internal */
export const Schema19ReasoningText$inboundSchema = z.object({
    type: z.literal("reasoning.text"),
    text: z.nullable(z.string()).optional(),
    signature: z.nullable(z.string()).optional(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema21$inboundSchema).optional(),
    index: z.number().optional(),
});
/** @internal */
export const Schema19ReasoningText$outboundSchema = z.object({
    type: z.literal("reasoning.text"),
    text: z.nullable(z.string()).optional(),
    signature: z.nullable(z.string()).optional(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema21$outboundSchema).optional(),
    index: z.number().optional(),
});
export function schema19ReasoningTextToJSON(schema19ReasoningText) {
    return JSON.stringify(Schema19ReasoningText$outboundSchema.parse(schema19ReasoningText));
}
export function schema19ReasoningTextFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema19ReasoningText$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema19ReasoningText' from JSON`);
}
/** @internal */
export const Schema19ReasoningEncrypted$inboundSchema = z.object({
    type: z.literal("reasoning.encrypted"),
    data: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema21$inboundSchema).optional(),
    index: z.number().optional(),
});
/** @internal */
export const Schema19ReasoningEncrypted$outboundSchema = z.object({
    type: z.literal("reasoning.encrypted"),
    data: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema21$outboundSchema).optional(),
    index: z.number().optional(),
});
export function schema19ReasoningEncryptedToJSON(schema19ReasoningEncrypted) {
    return JSON.stringify(Schema19ReasoningEncrypted$outboundSchema.parse(schema19ReasoningEncrypted));
}
export function schema19ReasoningEncryptedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema19ReasoningEncrypted$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema19ReasoningEncrypted' from JSON`);
}
/** @internal */
export const Schema19ReasoningSummary$inboundSchema = z.object({
    type: z.literal("reasoning.summary"),
    summary: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema21$inboundSchema).optional(),
    index: z.number().optional(),
});
/** @internal */
export const Schema19ReasoningSummary$outboundSchema = z.object({
    type: z.literal("reasoning.summary"),
    summary: z.string(),
    id: z.nullable(z.string()).optional(),
    format: z.nullable(Schema21$outboundSchema).optional(),
    index: z.number().optional(),
});
export function schema19ReasoningSummaryToJSON(schema19ReasoningSummary) {
    return JSON.stringify(Schema19ReasoningSummary$outboundSchema.parse(schema19ReasoningSummary));
}
export function schema19ReasoningSummaryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema19ReasoningSummary$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema19ReasoningSummary' from JSON`);
}
/** @internal */
export const Schema19$inboundSchema = z.union([
    z.lazy(() => Schema19ReasoningSummary$inboundSchema),
    z.lazy(() => Schema19ReasoningEncrypted$inboundSchema),
    z.lazy(() => Schema19ReasoningText$inboundSchema),
]);
/** @internal */
export const Schema19$outboundSchema = z
    .union([
    z.lazy(() => Schema19ReasoningSummary$outboundSchema),
    z.lazy(() => Schema19ReasoningEncrypted$outboundSchema),
    z.lazy(() => Schema19ReasoningText$outboundSchema),
]);
export function schema19ToJSON(schema19) {
    return JSON.stringify(Schema19$outboundSchema.parse(schema19));
}
export function schema19FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Schema19$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Schema19' from JSON`);
}
