// @openrouter/sdk/esm/models/schema5.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ceb3f15ca425
 */
import * as z from "zod/v4";
import * as openEnums from "../types/enums.js";
export const Schema5Enum = {
    Ai21: "AI21",
    AionLabs: "AionLabs",
    Alibaba: "Alibaba",
    Ambient: "Ambient",
    AmazonBedrock: "Amazon Bedrock",
    AmazonNova: "Amazon Nova",
    Anthropic: "Anthropic",
    ArceeAI: "Arcee AI",
    AtlasCloud: "AtlasCloud",
    Avian: "Avian",
    Azure: "Azure",
    BaseTen: "BaseTen",
    BytePlus: "BytePlus",
    BlackForestLabs: "Black Forest Labs",
    Cerebras: "Cerebras",
    Chutes: "Chutes",
    Cirrascale: "Cirrascale",
    Clarifai: "Clarifai",
    Cloudflare: "Cloudflare",
    Cohere: "Cohere",
    Crusoe: "Crusoe",
    DeepInfra: "DeepInfra",
    DeepSeek: "DeepSeek",
    Featherless: "Featherless",
    Fireworks: "Fireworks",
    Friendli: "Friendli",
    GMICloud: "GMICloud",
    Google: "Google",
    GoogleAIStudio: "Google AI Studio",
    Groq: "Groq",
    Hyperbolic: "Hyperbolic",
    Inception: "Inception",
    Inceptron: "Inceptron",
    InferenceNet: "InferenceNet",
    Infermatic: "Infermatic",
    Inflection: "Inflection",
    Liquid: "Liquid",
    Mara: "Mara",
    Mancer2: "Mancer 2",
    Minimax: "Minimax",
    ModelRun: "ModelRun",
    Mistral: "Mistral",
    Modular: "Modular",
    MoonshotAI: "Moonshot AI",
    Morph: "Morph",
    NCompass: "NCompass",
    Nebius: "Nebius",
    NextBit: "NextBit",
    Novita: "Novita",
    Nvidia: "Nvidia",
    OpenAI: "OpenAI",
    OpenInference: "OpenInference",
    Parasail: "Parasail",
    Perplexity: "Perplexity",
    Phala: "Phala",
    Relace: "Relace",
    SambaNova: "SambaNova",
    Seed: "Seed",
    SiliconFlow: "SiliconFlow",
    Sourceful: "Sourceful",
    StepFun: "StepFun",
    Stealth: "Stealth",
    StreamLake: "StreamLake",
    Switchpoint: "Switchpoint",
    Together: "Together",
    Upstage: "Upstage",
    Venice: "Venice",
    WandB: "WandB",
    Xiaomi: "Xiaomi",
    XAI: "xAI",
    ZAi: "Z.AI",
    FakeProvider: "FakeProvider",
};
/** @internal */
export const Schema5Enum$outboundSchema = openEnums.outboundSchema(Schema5Enum);
/** @internal */
export const Schema5$outboundSchema = z
    .union([Schema5Enum$outboundSchema, z.string()]);
export function schema5ToJSON(schema5) {
    return JSON.stringify(Schema5$outboundSchema.parse(schema5));
}
